src/librawspeed/decoders/IiqDecoder.cpp

```diff
@@ -242,6 +257,47 @@ void IiqDecoder::DecodePhaseOneC(const std::vector<IiqStrip>& strips,
     DecodeStrip(strip, width, height);
 }
 
+void IiqDecoder::CorrectPhaseOneC(ByteStream meta_data, uint32 split_row,
+                                  uint32 split_col) {
+  ByteStream meta_data_base(meta_data.getSubStream(0));
+
+  meta_data.skipBytes(8);
+  const uint32 bytes_to_entries = meta_data.getU32();
+  meta_data.skipBytes(bytes_to_entries - 12);
+  const uint32 entries_count = meta_data.getU32();
+  meta_data.skipBytes(4);
+
+  // this is how much is to be read for all the entries
+  ByteStream entries(meta_data.getStream(16 * entries_count));
```

Comment:
```diff
- ByteStream entries(meta_data.getStream(16 * entries_count));
+ ByteStream entries(meta_data.getStream(entries_count, 16));
```

Reply:
You read only 3x `U32` per entry there, why `16`?

Reply:
Copied from the previous metadata reading, this should be 12